dev/pre-commit

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+import sys
+import subprocess
+
+from flake8.main import git as flake8_run
+import pytest
+
+NO_COMMIT_MSG = 'NO COMMIT FOR YOU!\n'
+
+if __name__ == '__main__':
+    print('\nRUNNING FLAKE8')
+    flake8_errors = flake8_run.hook(strict=True)
+
+    print('\nRUNNING PYTEST')
+    pytest_errors = pytest.main()
+
+    problems = flake8_errors + pytest_errors
+
+    print(f'\nPytest error status {pytest_errors}, Flake8 error status {flake8_errors}.')
+
+    if problems:
+        print(NO_COMMIT_MSG)
+    else:
+        print('\nRUNNING FRONTEND TESTS')
+        frontend_errors = 0
+
+        try:
+            subprocess.check_call('cd penny_university_frontend && CI=true npm test', shell=True)
```

Comment:
Should we add frontend linting as well?

Reply:
Yes please! Just let me know what to add.